grails-app/domain/org/pih/warehouse/inventory/InventoryItem.groovy

```diff
@@ -127,4 +129,40 @@ class InventoryItem implements Serializable {
     }
 
 
+    /**
+     * Get the first receiving date for this inventory item in given bin location at the given location.
+     *
+     * @param locationId
+     * @return
+     */
+    Date firstReceivingDate(String locationId, Location binLocatiom) {
+        def inventory = Location.get(locationId).inventory
+        def date
+        if (binLocatiom) {
+            date = TransactionEntry.executeQuery("""
+						select 
+						     min(t.transactionDate)
+						from TransactionEntry as te
+						left join te.transaction as t
+						where te.binLocation = :binLocation
+						and te.inventoryItem = :inventoryItem
+						and t.inventory = :inventory
+						and t.transactionType in (:transactionTypes)
+						""", [inventory: inventory, binLocation: binLocatiom, inventoryItem: this, transactionTypes: TransactionType.get(Constants.TRANSFER_IN_TRANSACTION_TYPE_ID)]).first()
+        } else {
```

Comment:
Cant you just do in part above: 
> binLocation: binLocatiom ?: null

Reply:
Or actually, i think there should be "where te.binLocaiton is null" instead of " = null"

Reply:
And you could do simple string concatenation instead of this second huge else part.

Reply:
@jmiranda We cannot pass null argument to executeQuery method because of NPE. I have tried creating query 
`String query = """ 
select 
min(t.transactionDate) 
from TransactionEntry as te 
left join te.transaction as t 
where ${binLocation ? "te.binLocation = ${binLocation}" : 'te.binLocation is null'} 
and te.inventoryItem = ${this} and t.inventory = '${inventory}' 
and t.transactionType in ('${TransactionType.get(Constants.TRANSFER_IN_TRANSACTION_TYPE_ID)}') """ 
        def date = TransactionEntry.executeQuery(query).first()`
but it also doesn't work. Do you have any suggestions on how should this one be implemented?

Reply:
@DominikaHatala I'll take a look (note: will be assigning ticket to me to remind me to work on it).